commcare-cloud/commcare_cloud/commands/ansible/ansible_playbook.py

```diff
@@ -229,3 +235,198 @@ def run(self, args, unknown_args):
+            )
+        )
+
+    def get_inventory_group_for_service(self, service, service_group):
+        return self.INVENTORY_GROUP_FOR_SERVICE.get(service, service_group)
+
+    def run_status_for_service(self, service_group, args, unknown_args):
+        for service in self.run_for_services(service_group, args):
+            if service == "redis":
+                args.shell_command = "redis-cli ping"
+            else:
+                args.shell_command = "service %s status" % self.SERVICE_PACKAGES_FOR_SERVICE.get(service, service)
+            args.inventory_group = self.get_inventory_group_for_service(service, args.service_group)
+            RunShellCommand(self.parser).run(args, unknown_args)
+
+    def run_ansible_module_for_service_group(self, service_group, args, unknown_args, inventory_group=None):
+        for service in self.SERVICES[service_group]:
+            action = args.action
+            state = self.DESIRED_STATE_FOR_ACTION[action]
+            args.inventory_group = (
+                inventory_group or
+                self.get_inventory_group_for_service(service, service_group))
+            args.module = 'service'
+            args.module_args = "name=%s state=%s" % (
+                self.ANSIBLE_MODULE_FOR_SERVICE.get(service, service),
+                state)
+            RunAnsibleModule(self.parser).run(
+                args,
+                unknown_args
+            )
+
+    def run_for_services(self, service_group, args):
+        if args.only:
+            return args.only.split(',')
+        else:
+            return self.SERVICES[service_group]
+
+    def run_for_es(self, args, unknown_args):
+        action = args.action
+        if action == "restart":
+            RestartElasticsearch(self.parser).run(args, unknown_args)
+        else:
+            self.run_ansible_module_for_service_group("es", args, unknown_args)
+
+    def run_for_riakcs(self, args, unknown_args):
+        tags = []
+        action = args.action
+        state = self.DESIRED_STATE_FOR_ACTION[action]
+        args.playbook = "service_playbooks/riakcs.yml"
+        run_for_services = self.run_for_services('riakcs', args)
+        if args.only:
+            # for options to act on certain services create tags
+            for service in run_for_services:
+                if service:
+                    tags.append("%s_%s" % (action, service))
+            if tags:
+                unknown_args.append('--tags=%s' % ','.join(tags),)
+        unknown_args.extend(['--extra-vars', "desired_state=%s desired_action=%s" % (state, action)])
+        # ToDo: use this with when in the playbook instead of tags
+        # currently its running riak even when just riak-cs is ran
+        # but skips riak when running for only stanchion
+        # as of now it looks like
+        # --extra-vars 'desired_services=['"'"'riak-cs'"'"', '"'"'stanchion'"'"']'
+        unknown_args.extend(['--extra-vars', "desired_services=%s" % run_for_services])
```

Comment:
I can't see where the `desired_services` is used anywhere in the playbooks. Was this from an older commit you've since changed away from / can you just remove these lines at this point?

Reply:
@dannyroberts i have this ToDo above this which shares where this can be used. I want to replace the playbook to use `desired_services` instead of relying on tags. Currently facing a challenge with how its making this as a command line argument so need to look into that.

Reply:
Ah, okay. Well I think what you would do in the command line is `... --extra-vars '{"desired_services": ["riak-cs", "stanchion"]}' ...`—i.e. if you need any part of it to be JSON and not just a string value then I believe you have to pass in the entire arg as a JSON dict.

If you don't end up using this in this PR, I suggest removing it and cleaning things up first, and then refactoring in a later PR. Leaving code in from an idea that didn't end up panning out is in my opinion (and I think generally recognized as) an anti-pattern: if it's not working now, then it's unlikely that having it in the code for later when/if you do decide to use it will be more helpful than starting from scratch looking at your previous code as a reference, and it's all saved in the commit history. So I'd still suggest removing this now, even if you think you're going to come back to it soon.

Reply:
@dannyroberts i could not get this done.. struggled due to my inexperience with ansible. Will get to this later after learning a bit more about ansible. 
removed this section.